vickev/howdypix
# Mock graphQL server in the client

## Description

Add fixture mocking in the `webapp` application for tests: it will mock all the `/graphql` queries with a set of fixtures defined in `server/mock/fixtures`. 

When you start the server with `NODE_ENV=test`, you can then go to `?fixture-set=<file>`. For now, there only is `full` and `empty`. It allows to load different set of fixtures without restarting the server.

This capability will be used for the Cypress tests - there won't be any `server` running to run the tests. This will come in a future PR.





## How to test?

- Check-out this branch.
- Go to `apps/webapp` and run the server: `NODE_ENV=test yarn dev`
- Go to `http://localhost:3000/?fixture-set=full` and `http://localhost:3000/?fixture-set=empty` => it should load different fixtures.

## Checklist
